File: test-suite-thymeleaf-fieldset/src/test/java/io/micronaut/views/fields/thymeleaf/BookAuthorSave.java
package io.micronaut.views.fields.thymeleaf;

import io.micronaut.serde.annotation.Serdeable;
import io.micronaut.views.fields.annotations.InputHidden;
import io.micronaut.views.fields.annotations.Select;
import jakarta.validation.constraints.NotNull;

@Serdeable
public record BookAuthorSave(@NotNull @InputHidden Long bookId,
                             @NotNull @Select(fetcher = AuthorFetcher.class) Long authorId) {
}

File: test-suite-python/src/test/python/micronaut/views/docs/fieldset/BookAuthorSave.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotNull
from micronaut.serde.annotation import Serdeable
from micronaut.views.fields.annotations import InputHidden, Select

from .AuthorFetcher import AuthorFetcher


# tag::clazz[]
@Serdeable
@dataclass
class BookAuthorSave:
    bookId: Annotated[int, NotNull, InputHidden]
    authorId: Annotated[int | None, NotNull, Select(fetcher=AuthorFetcher)]
# end::clazz[]
